panther_lights/test/test_apa102.cpp

```diff
@@ -31,52 +50,121 @@ class APA102Wrapper : public panther_lights::apa102::APA102
 class TestAPA102 : public testing::Test
 {
 protected:
-  TestAPA102() { apa102_ = std::make_unique<APA102Wrapper>("/dev/spidev0.0"); }
-
+  TestAPA102();
   ~TestAPA102() { apa102_.reset(); }
 
+  std::shared_ptr<MockSPIDevice> spi_device_;
   std::unique_ptr<APA102Wrapper> apa102_;
 };
 
-TEST_F(TestAPA102, PortsAvailable)
+TestAPA102::TestAPA102()
+{
+  spi_device_ = std::make_shared<MockSPIDevice::NiceMock>();
+
+  apa102_ = std::make_unique<APA102Wrapper>(spi_device_, kMockDeviceName);
+}
+
+TEST_F(TestAPA102, TestInitialization) { EXPECT_TRUE(spi_device_ != nullptr); }
+
+TEST(TestInitialization, InvalidDevices)
+{
+  auto spi_device = std::make_shared<MockSPIDevice::NiceMock>();
+
+  // Return -1 to simulate failed device opening
+  ON_CALL(*spi_device, Open(kMockDeviceName)).WillByDefault(testing::Return(-1));
+
+  EXPECT_THROW(std::make_unique<APA102Wrapper>(spi_device, kMockDeviceName);
+               , std::ios_base::failure);
```

Comment:
```suggestion
  EXPECT_THROW(APA102Wrapper(spi_device, kMockDeviceName)
  , std::ios_base::failure);
```
Don't think we need pointer here. Also, is this semicolon necessary here?
Applies to other initialization tests.